src/figuresClasses.ts

```diff
@@ -1,11 +1,85 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+function checkOnError(errorMessage: string, ...sides: number[]): void {
+  if (sides.some((side) => side <= 0)) {
+    throw new Error(errorMessage);
+  }
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const triangleErrorMessage =
+      'Invalid triangle sides: ensure all sides are positive.';
 
-export class Circle implements Figure {}
+    checkOnError(triangleErrorMessage, a, b, c);
+
+    const sideOfTriangle = [a, b, c].sort((a1, b1) => b1 - a1);
+    const condition =
+      sideOfTriangle[0] >= sideOfTriangle[1] + sideOfTriangle[2];
+
+    if (condition) {
+      throw new Error(triangleErrorMessage);
+    }
+  }
+
+  getArea(): number {
+    const s = (1 / 2) * (this.a + this.b + this.c);
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.round(area * 100) / 100;
+  }
+}
 
-export class Rectangle implements Figure {}
+export class Circle implements Figure {
+  public shape: Shape = 'circle';
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    const circleErrorMessage =
+      'Invalid radius: radius must be a positive number.';
```

Comment:
The error message for an invalid circle radius is good as it provides a clear description of the error. However, it could also suggest the correct condition, e.g., 'Invalid radius: radius must be a positive number greater than zero.'